aspnetcore/performance/caching/memory.md

```diff
@@ -1,164 +1,111 @@
 
-**`Get`**
+## Using `IMemoryCache`
 
-`Get` will return the value if it exists, but otherwise returns `null`.
+In-memory caching is a *service* that is referenced from your app using [Dependency Injection](../../fundamentals/dependency-injection.md). Call `AddMemoryCache` in `ConfigureServices`:
 
-**`TryGet`**
+[!code-csharp[Main](memory/sample/WebCache/Startup.cs?highlight=8,9)] 
 
-`TryGet` will assign the cached value to an `out` parameter and return true if the entry exists. Otherwise it returns false.
-
-Use the `Set` method to write to the cache. `Set` accepts the key to use to look up the value, the value to be cached, and a set of `MemoryCacheEntryOptions`. The `MemoryCacheEntryOptions` allow you to specify absolute or sliding time-based cache expiration, caching priority, callbacks, and dependencies. These options are detailed below.
-
-The sample code (shown below) uses the `SetAbsoluteExpiration` method on `MemoryCacheEntryOptions` to cache greetings for one minute.
-
-[!code-csharp[Main](memory/sample/src/CachingSample/Middleware/GreetingMiddleware.cs?highlight=7,10,16-18&range=30-58)]
-
-In addition to setting an absolute expiration, a sliding expiration can be used to keep frequently requested items in the cache:
-
-```csharp
-// keep item in cache as long as it is requested at least
-// once every 5 minutes
-new MemoryCacheEntryOptions()
-  .SetSlidingExpiration(TimeSpan.FromMinutes(5))
-```
-
-To avoid having frequently-accessed cache entries growing too stale (because their sliding expiration is constantly reset), you can combine absolute and sliding expirations:
-
-```csharp
-// keep item in cache as long as it is requested at least
-// once every 5 minutes...
-// but in any case make sure to refresh it every hour
-new MemoryCacheEntryOptions()
-  .SetSlidingExpiration(TimeSpan.FromMinutes(5))
-  .SetAbsoluteExpiration(TimeSpan.FromHours(1))
-```
-
-By default, an instance of `MemoryCache` will automatically manage the items stored, removing entries when necessary in response to memory pressure in the app. You can influence the way cache entries are managed by setting their `CacheItemPriority` when adding the item to the cache. For instance, if you have an item you want to keep in the cache unless you explicitly remove it, you would use the `NeverRemove` priority option:
-
-```csharp
-// keep item in cache indefinitely unless explicitly removed
-new MemoryCacheEntryOptions()
-  .SetPriority(CacheItemPriority.NeverRemove))
-```
+Request the `IMemoryCache` instance in the constructor:
 
-When you do want to explicitly remove an item from the cache, you can do so easily using the `Remove` method:
+```c#
+  public class HomeController : Controller
+  {
 
-```csharp
-cache.Remove(cacheKey);
+      public HomeController(IMemoryCache memoryCache)
+      {
+          _memoryCache = memoryCache;
+      }
 ```
 
-## Cache Dependencies and Callbacks
+`IMemoryCache` requires NuGet package "Microsoft.Extensions.Caching.Memory".
```

Comment:
My preference would be to remove it too, but code comments are not localized and we're having the docs localized into 10 languages next month.

I made changes to everything you mentioned except this and ephemeral.